src/ScreenshotManager.vala

```diff
@@ -167,15 +167,22 @@ namespace Gala
 
 		static async bool save_image (Cairo.ImageSurface image, string filename, out string used_filename)
 		{
-			if (!Path.is_absolute (filename)) {
-				var path = find_target_path ();
-				if (!filename.has_suffix (".png")) {
-					used_filename = Path.build_filename (path, filename.concat (".png"), null);
-				} else {
-					used_filename = Path.build_filename (path, filename, null);
+			used_filename = filename;
+
+			// We only alter non absolute filename because absolute
+			// filename is used for temp clipboard file and shouldn't be changed
+			if (!Path.is_absolute (used_filename)) {
+				if (!used_filename.has_suffix (".png")) {
+					used_filename = used_filename.concat (".png");
 				}
-			} else {
-				used_filename = filename;
+
+				var scale_factor = InternalUtils.get_ui_scaling_factor ();
+				if (scale_factor > 1) {
+					used_filename = used_filename.splice (-4, -4, "@%ix".printf (scale_factor));
```

Comment:
At this point the .png string should be a constant and here instead of using hardcodes, use the length of the constant.

Reply:
Good point. Done.